File: 2024.2/LAB/01/Recursivo.java
import java.util.Scanner;

public class Recursivo {
    public static void main(String[] args) {
        Scanner myObj = new Scanner(System.in);

        while (true) {
            String text = myObj.nextLine();

            if (text.equals("FIM"))
                break;

            int upperLetters = countUpperCase(text, text.length() - 1);
            System.out.println(upperLetters);
        }
        myObj.close();
    }

    public static int countUpperCase(String text, int index) {
        if (index < 0) {
            return 0;
        }

        int count = Character.isUpperCase(text.charAt(index)) ? 1 : 0;

        return count + countUpperCase(text, index - 1);
    }
}

// javac Recursivo.java && java Recursivo < 1_pub.in > minhaSaida.out && diff 1_pub.out minhaSaida.out

File: 2024.2/LAB/01/Recursivo.c
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>
#include <string.h>
#include <stdbool.h>

int countUpperCase(char* str, int index) {
    if (index == strlen(str)) {
        return 0;
    }

    int count = (isupper(str[index])) ? 1 : 0;

    return count + countUpperCase(str, index + 1);
}

int main() {
    char myString[100];
    int upper;

    while (true) {
        scanf(" %[^\n]", myString);

        if (strcmp(myString, "FIM") == 0)
            break;

        upper = countUpperCase(myString, 0);
        printf("%d\n", upper);
    }

    return 0;
}

// gcc Recursivo.c -o minhaSaida && ./minhaSaida